etc/sysctl.conf

```
@@ -1,90 +1,92 @@
+net.inet.tcp.sendbuf_inc=131072          # Incremental send buffer growth step (128 KB)
+net.inet.tcp.sendbuf_max=16777216        # Max send buffer size (16 MB)
+net.inet.tcp.syncache.rexmtlimit=0       # Disable SYN cache retransmit limit
+net.inet.tcp.syncookies=0                # Disable SYN cookies (only needed during SYN flood)
+net.inet.tcp.tso=0                       # Disable TCP Segmentation Offload (can cause driver issues)
 
-net.inet6.icmp6.nodeinfo=0
-net.inet6.icmp6.rediraccept=0
-net.inet6.ip6.accept_rtadv=0
-net.inet6.ip6.auto_linklocal=0
-net.inet6.ip6.maxfragpackets=0
-net.inet6.ip6.maxfrags=0
-net.inet6.ip6.prefer_tempaddr=1
-net.inet6.ip6.use_tempaddr=1
-net.inet6.ip6.use_deprecated=0
+net.inet.udp.blackhole=1                 # Drop UDP packets to closed ports (no ICMP unreachable)
 
-net.local.dgram.maxdgram=65535
-net.local.dgram.recvspace=65535
+net.inet6.icmp6.nodeinfo=0               # Disable ICMPv6 Node Information responses
+net.inet6.icmp6.rediraccept=0            # Reject ICMPv6 redirects
+net.inet6.ip6.accept_rtadv=0             # Disable IPv6 Router Advertisements (no autoconf)
+# read-only now, use ifconfig instead ->
+net.inet6.ip6.auto_linklocal=0           # Do not auto-generate link-local addresses
+net.inet6.ip6.maxfragpackets=0           # Disable IPv6 fragment reassembly
+net.inet6.ip6.maxfrags=0                 # Limit fragment buffer usage
+net.inet6.ip6.prefer_tempaddr=1          # Prefer temporary (privacy) addresses
+net.inet6.ip6.use_tempaddr=1             # Enable IPv6 temporary address generation
+net.inet6.ip6.use_deprecated=0           # Do not use deprecated IPv6 addresses
 
-security.bsd.hardlink_check_gid=1
-security.bsd.hardlink_check_uid=1
-security.bsd.see_other_gids=0
-security.bsd.see_other_uids=0
-security.bsd.stack_guard_page=1
-security.bsd.unprivileged_proc_debug=0
-security.bsd.unprivileged_read_msgbuf=0
+net.local.dgram.maxdgram=65535           # Max datagram size for UNIX domain sockets
+net.local.dgram.recvspace=65535          # Receive buffer size for UNIX domain sockets
 
-vfs.zfs.vdev.async_read_max_active=6
-vfs.zfs.vdev.sync_read_max_active=20
-vfs.read_max=128
+security.bsd.hardlink_check_gid=1        # Prevent hardlinking across GIDs (privilege restriction)
+security.bsd.hardlink_check_uid=1        # Prevent hardlinking across UIDs (mitigate privilege escalation)
+security.bsd.see_other_gids=0            # Hide processes of other GIDs (isolation)
+security.bsd.see_other_uids=0            # Hide processes of other UIDs (privacy)
+security.bsd.stack_guard_page=1          # Enable stack guard page (stack overflow protection)
+security.bsd.unprivileged_proc_debug=0   # Disallow ptrace/debug on other users’ processes
+security.bsd.unprivileged_read_msgbuf=0  # Prevent non-root users from reading kernel message buffer
+
+vfs.zfs.bclone_enabled=1                 # Allows shallow copies of data blocks (introduced in OpenZFS 2.2). 
```

Take a moment to identify the need of this setting please.

---

I prefer not enabling this for now, as it could cause unexpected problems
```suggestion
vfs.zfs.bclone_enabled=0                 # Allows shallow copies of data blocks (introduced in OpenZFS 2.2). 
```